.github/workflows/node.js.yml

```diff
@@ -6,19 +6,107 @@ name: Node.js CI
 on: [push, pull_request]
 
 jobs:
-  build:
-
+  common:
+    name: Common tests (Node.js v${{ matrix.node-version }})
     runs-on: macos-latest
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 16.x]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: bahmutov/npm-install@v1
+        with:
+          useRollingCache: true
+      - run: npm run test:common
 
+  ios-simulator_unit:
+    name: ios-simulator unit tests (Node.js v${{ matrix.node-version }} / iOS v${{ matrix.ios-version }} / Xcode v${{ matrix.xcode-version }} / ${{ matrix.device-name }})
+    runs-on: macos-latest
     strategy:
       matrix:
         node-version: [12.x, 14.x, 16.x]
+        include:
+          - device-name: iPhone 12
+            xcode-version: 12.4
+            ios-version: 14.4
+    steps:
+      - uses: actions/checkout@v2
+      - uses: bahmutov/npm-install@v1
+        with:
+          useRollingCache: true
+      - name: Configure Homebrew cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/Library/Caches/Homebrew/applesimutils--*
+            ~/Library/Caches/Homebrew/downloads/*--applesimutils-*
+          key: brew-${{ hashFiles('.github/brew-formulae') }}
+          restore-keys: brew-
+      - name: Update Homebrew
+        run: |
+          brew tap wix/brew
+          brew update --preinstall
+          cat "$(brew --repository)/Library/Taps/wix/homebrew-brew/applesimutils.rb" > .github/brew-formulae
+      - name: Install Homebrew dependencies
+        env:
+          HOMEBREW_NO_AUTO_UPDATE: 1
+        run: brew install applesimutils
```

Comment:
do we really need this all stuff to run unit tests?

Reply:
probably not, I'll remove them and see

Reply:
It doesn't appear they are needed.  removed